playbooks/common/openshift-cluster/upgrades/files/ensure_system_units_have_version.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+set -e
+
+SERVICE_TYPE=$1
+DEPLOYMENT_TYPE=$2
+VERSION="v${3}"
+
+add_image_version_to_sysconfig () {
+    unit_name=$2
+    sysconfig_file=/etc/sysconfig/${unit_name}
+
+    if ! grep IMAGE_VERSION ${sysconfig_file}; then
+        sed -i "/CONFIG_FILE/a IMAGE_VERSION=${1}" ${sysconfig_file}
+    else
+        sed -i "s/\(IMAGE_VERSION=\).*/\1${1}/" ${sysconfig_file}
+    fi
+}
+
+add_image_version_to_unit () {
```

Comment:
I really struggled with what appeared to be cached values returned by openshift_facts when upgrades were in progress.  I'll have to see if I can come up with a reproducer outside of the upgrade.  Part of me just wants to get this to QE officially as soon as possible so they can start finding serious bugs.